### PR TITLE
use defaults utility at relationships.

### PR DESCRIPTION
**Pull request by mshima**
Related to https://github.com/jhipster/generator-jhipster/issues/23917.

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Comment by mshima** [comment]
This doesn’t fix the problem, but the api for loading relationship is there.
Fetching not ownerSide is only useful for view, unless we are going to cascade, so we probably need to add a flag for write/read.

**Comment by hide212131** [comment]
I've reviewed the generated code and have a general understanding of the concept.

File: generators/base-application/support/prepare-entity.mts [hunk 1]
```diff
@@ -616,14 +623,15 @@ function preparePostEntityCommonDerivedPropertiesNotTyped(entity: any) {
...
+
+    mutateData(relationship, {
+      bagRelationship: relationship.collection,
+      relationshipEagerLoad: () =>
         relationship.bagRelationship ||
         entity.eagerLoad ||
         // Fetch relationships if otherEntityField differs otherwise the id is enough
-        (relationship.ownerSide && relationship.otherEntity.primaryKey.name !== relationship.otherEntityField);
-    }
+        relationship.otherEntity.primaryKey.name !== relationship.otherEntityField,
```
Comment (mshima) [inline generators/base-application/support/prepare-entity.mts:633]:
```suggestion
        relationship.ownerSide && relationship.otherEntity.primaryKey.name !== relationship.otherEntityField),
```

File: generators/base-application/support/prepare-entity.mts [hunk 2]
```diff
@@ -616,14 +623,15 @@ function preparePostEntityCommonDerivedPropertiesNotTyped(entity: any) {
       relationship.relationshipEagerLoad = false;
       return;
     }
-    relationship.bagRelationship = relationship.ownerSide && relationship.collection;
-    if (relationship.relationshipEagerLoad === undefined) {
-      relationship.relationshipEagerLoad =
+
+    mutateData(relationship, {
+      bagRelationship: relationship.collection,
```
Comment (mshima) [inline generators/base-application/support/prepare-entity.mts:628]:
```suggestion
      bagRelationship: relationship.ownerSide && relationship.collection,
```

File: generators/base-application/support/prepare-entity.mts [hunk 3]
```diff
@@ -616,14 +623,15 @@ function preparePostEntityCommonDerivedPropertiesNotTyped(entity: any) {
...
+
+    mutateData(relationship, {
+      bagRelationship: relationship.ownerSide && relationship.collection,
+      relationshipEagerLoad: () =>
         relationship.bagRelationship ||
         entity.eagerLoad ||
         // Fetch relationships if otherEntityField differs otherwise the id is enough
-        (relationship.ownerSide && relationship.otherEntity.primaryKey.name !== relationship.otherEntityField);
-    }
+        relationship.ownerSide && relationship.otherEntity.primaryKey.name !== relationship.otherEntityField),
```
Comment (mshima) [inline generators/base-application/support/prepare-entity.mts:633]:
```suggestion
        (relationship.ownerSide && relationship.otherEntity.primaryKey.name !== relationship.otherEntityField),
```

**Review by DanielFran — APPROVED** [review]

**Pull request merged**